### PR TITLE
[FLINK-35176][Connector/JDBC] Support property authentication connection for JDBC catalog & dynamic table

### PR DESCRIPTION
**Pull request by RocMarshal**
- Support property authentication connection for JDBC catalog & dynamic table

### CODE REVIEW

**Comment by RocMarshal** [comment]
blocked by https://github.com/apache/flink-connector-jdbc/pull/115

**Comment by RocMarshal** [comment]
HI, @caicancai @GOODBOY008 Could you help have a review if you had the free time ? thx a lot.

File: flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/JdbcCatalog.java [hunk 1]
```diff
@@ -77,17 +81,42 @@ public JdbcCatalog(
...
+                userClassLoader,
+                catalogName,
+                defaultDatabase,
+                baseUrl,
+                compatibleMode,
+                getBriefAuthProperties(username, pwd));
+    }
+
+    /**
+     * Creates a JdbcCatalog.
```
Comment (caicancai) [inline flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/JdbcCatalog.java:94]: creates？

Comment (caicancai) [reply flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/JdbcCatalog.java:94]: I mean why the comment is Creates a JdbcCatalog, not Create a JdbcCatalog

**Comment by RocMarshal** [comment]
Thanks @caicancai for the review.
Could @1996fanrui you help take a look if you had the free time ? 
Thank you~

**Comment by RocMarshal** [comment]
Hi, @eskabetxe Would you mind having take a look if you had the free time ? Many thx!

File: flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java [hunk 2]
```diff
@@ -88,32 +92,49 @@ public abstract class AbstractJdbcCatalog extends AbstractCatalog {
...
 
         JdbcCatalogUtils.validateJdbcUrl(baseUrl);
 
         this.userClassLoader = userClassLoader;
-        this.username = username;
-        this.pwd = pwd;
         this.baseUrl = baseUrl.endsWith("/") ? baseUrl : baseUrl + "/";
         this.defaultUrl = this.baseUrl + defaultDatabase;
+        this.connectionProperties = Preconditions.checkNotNull(connectionProperties);
+        if (StringUtils.isNullOrWhitespaceOnly(connectionProperties.getProperty(USER_KEY))) {
```
Comment (eskabetxe) [inline flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/catalog/AbstractJdbcCatalog.java:132]: we should maintain the current checks
```
checkArgument(!StringUtils.isNullOrWhitespaceOnly(connectionProperties.getProperty(USER_KEY)));
checkArgument(!StringUtils.isNullOrWhitespaceOnly(connectionProperties.getProperty(PASSWORD_KEY)));
```

**Comment by RocMarshal** [comment]
Thanks @eskabetxe for the comments. Updated.

**Review by eskabetxe — APPROVED** [review]
LGTM

**Review by 1996fanrui — COMMENTED** [review]
Thanks @RocMarshal for the contribution, and @eskabetxe @caicancai  for the review!

Merging~

**Pull request merged**